projects/storefrontlib/src/cms-components/product/product-list/container/product-scroll/product-scroll.component.ts

```diff
@@ -164,7 +195,24 @@ export class ProductScrollComponent implements OnDestroy {
     return false;
   }
 
-  ngOnDestroy(): void {
-    this.subscription.unsubscribe();
+  ngAfterViewChecked(): void {
+    // if auto scroll positon exists
+    if (
+      this.productListComponentService.autoScrollPosition[0] !== 0 ||
+      this.productListComponentService.autoScrollPosition[1] !== 0
+    ) {
+      // 1. no scroll done then do the scroll.
+      // 2. even the scroll done, but we want another scroll if there is a view change during 300ms.
+      if (!this.doneAutoScroll || Date.now() - this.lastScrollTime < 300) {
```

Comment:
instead of making it scroll, and have the user see it happen with all the loaded spinner before going to the position, we can maybe have it load before scrolling.